text/unified-reduction-strategies.md

```diff
@@ -0,0 +1,83 @@
+- Title: A unified view at reduction strategies with extended unfolding control 
+
+- Drivers: Hugo Herbelin
+
+----
+
+# Summary and motivation
+
+Each user-level, compilation-free, reduction strategy of Coq, `simpl`,
+`cbn`, `cbv`, `lazy`, come, in addition to a reduction order
+(call-by-name, call-by-value, call-by-need), with its own
+specificities, such as:
+
+- refolding (co)fixpoints (in `simpl` and `cbn`)
+- local specification of the kind of reduction to consider (`cbn`, `cbv`, `lazy`)
+- specification of a subterm to reduce (`simpl pattern`)
+- some static parameterization of the unfolding (argument flags `simpl never`, `simpl nonmeta`, `!`, `/`)
+
+The proposal is to go in the direction of one unified entry point to
+reduction strategies eventually supporting all combinations of the
+features currently provided only by one or the other entry point, thus
+gaining both in expressiveness and easiness to document. At the
+current time, the focus is however only on a unified syntax.
+
+Additionally, the proposal is intended to complement CEP coq/ceps#35
+(more fine-tuning of reduction) and CEP coq/ceps#84 (support for
+symbol sets, hereafter called unfol bases).
+
+# Detailed design
+
+We propose to elaborate a unified syntax for all four mentioned user-level reduction strategies on top of the following:
+
+```text
+red_expr   ::= red_name pattern_opt flags_opt
+red_name   ::= "cbv" | "cbn" | "lazy" | "simpl"
+pattern    ::= CONSTR
+flags      ::= "with" head_flag_opt flag_list unfold_bases
```

Comment:
why is head_flag separate? Is there something wrong with writing eg `with beta head`?

Reply:
I don't remember if there was a technical reason or just because I was thinking at `head` as a modifier of a different nature than the flags.